sdk/storage/azure-storage-blob/azure/storage/blob/_shared/streams.py

```diff
@@ -532,38 +529,36 @@ def _read_message_header(self) -> None:
         if self.message_version == 1:
             message_length = int.from_bytes(self._read_from_inner(8), 'little')
             if message_length != self.message_length:
-                raise StructuredMessageError(f"Structured message length {message_length} "
-                                             f"did not match content length {self.message_length}")
+                raise ValueError(f"Structured message length {message_length} "
+                                 f"did not match content length {self.message_length}")
 
             self.flags = StructuredMessageProperties(int.from_bytes(self._read_from_inner(2), 'little'))
             self.num_segments = int.from_bytes(self._read_from_inner(2), 'little')
 
             self._message_offset += StructuredMessageConstants.V1_HEADER_LENGTH
 
         else:
-            raise StructuredMessageError(f"The structured message version is not supported: {self.message_version}")
+            raise ValueError(f"The structured message version is not supported: {self.message_version}")
 
     def _read_message_footer(self) -> None:
         # Sanity check: There should only be self._message_footer_length (could be 0) bytes left to consume.
         # If not, it is likely the message header contained incorrect info.
         if self.message_length - self._message_offset != self._message_footer_length:
-            raise StructuredMessageError("Invalid structured message data detected.")
+            raise ValueError("Invalid structured message data detected.")
 
         if StructuredMessageProperties.CRC64 in self.flags:
             message_crc = self._read_from_inner(StructuredMessageConstants.CRC64_LENGTH)
 
             if self._message_crc64 != int.from_bytes(message_crc, 'little'):
-                raise StructuredMessageError("CRC64 mismatch detected in message trailer. "
-                                             "All data read should be considered invalid.")
+                raise ValueError("CRC64 mismatch detected in message trailer. "
+                                 "All data read should be considered invalid.")
 
         self._message_offset += self._message_footer_length
 
     def _read_segment_header(self) -> None:
-        self._assert_remaining_length(self._segment_header_length)
```

Comment:
This comment applies broadly-- any reason we don't want to have this check anymore? I see the function being completely removed and it's logic not being replaced in all the callsites-- just wondering if it was a wrong check or if you deem it unnecessary?

Reply:
Good question! I removed `_assert_remaining_length` because the changes I made to `_read_from_inner` handle the same check but better.

What I used to do is before reading a section I would check to see if the response body had enough data for that whole section purely based on the `content_length` that the service returned and how much data we think we'd read. This wasn't always reliable. The new check checks that EVERY time we read from the inner stream (the network stream for the response body in the live case), we get back the amount of data we requested, because all of our reads from the inner stream expect a certain length (segment header size, crc size, etc.).